clients/js/src/auth.ts

```diff
@@ -130,6 +133,85 @@ class BasicAuthClientAuthProvider implements ClientAuthProvider {
     }
 }
 
+class TokenAuthCredentials implements AbstractCredentials<SecretStr> {
+    private readonly credentials: SecretStr;
+
+    constructor(_creds: string) {
+        this.credentials = new SecretStr(_creds)
+    }
+
+    getCredentials(): SecretStr {
+        return this.credentials;
+    }
+}
+
+export class TokenCredentialsProvider implements ClientAuthCredentialsProvider<TokenAuthCredentials> {
```

Comment:
do you think it makes sense to have this as a separate class since its only used in one place?

Reply:
If you mean the `TokenCredentialsProvider` class, that is part of the abstraction. The intent here is that auth providers are decoupled from the credentials providers. For our initial use case, this is pretty straightforward: take config value or env var, and there is a world where tokens can be fetched via some KMS or vault.
Another use case, in particular for the TokenCredentialsProvider class, I see is when we go to multi-user authZ where the user token (not to be confused with the current impl which is not a user token but rather a global static token) is stored as env var and used by a different auth provider (e.g. UserTokenClientAuthProvider).

Bottom line: Let's acknowledge this as a potential technical debt we can quickly repay by merging the constructor with TokenClientAuthProvider. If we do not need the CredentialsProvider abstraction (further down the road), we can easily merge constructors for a simpler solution.